src/rhc.erl

```diff
@@ -203,13 +209,90 @@ fetch(Rhc, QueueName) ->
             {error, Error}
     end.
 
+-spec fetch(rhc(), binary(), internal|internal_aaehash) ->
+                {ok, queue_empty}|
+                {error, term()}|
+                {ok|crc_wonky,
+                    {deleted, term(), binary()}|
+                        binary()|
+                        {deleted, term(), binary(), non_neg_integer(), non_neg_integer()}|
+                        {binary(), non_neg_integer(), non_neg_integer()}}.
+fetch(Rhc, QueueName, internal) ->
+    fetch(Rhc, QueueName);
+fetch(Rhc, QueueName, internal_aaehash) ->
+    QParams = [{object_format, internal_aaehash}],
+    URL = fetch_url(Rhc, QueueName, QParams),
+    case request(get, URL, ["200"], [], [], Rhc) of
+        {ok, _Status, _Headers, Body} ->
+            case Body of
```

Comment:
Here is the hard coded decoding of what was encoded in riak_kv_wm_queue... This seems to me a bit error prone, rather would I have one place in which decode and encode are handled. But I agree, having that in this client repo would not be most logic. Having it is riak+kv would be wrong, because then there is a circular dependency... Possibly in leveled_aae or so?

Reply:
Acknowledged - see https://github.com/basho/riak_kv/pull/1792